hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -158,15 +158,20 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tablePath: String,
       file.partitionValues match {
         // Snapshot or incremental queries.
         case fileSliceMapping: HoodiePartitionFileSliceMapping =>
-          val filegroupName = FSUtils.getFileIdFromFilePath(sparkAdapter
+          val fileGroupName = FSUtils.getFileIdFromFilePath(sparkAdapter
             .getSparkPartitionedFileUtils.getPathFromPartitionedFile(file))
-          fileSliceMapping.getSlice(filegroupName) match {
-            case Some(fileSlice) if !isCount && (requiredSchema.nonEmpty || fileSlice.getLogFiles.findAny().isPresent) =>
```

Comment:
why remove the `requiredSchema.nonEmpty` check.

Reply:
> why remove the `requiredSchema.nonEmpty` check.

<img width="681" alt="image" src="https://github.com/user-attachments/assets/9d39ebe7-4a22-4d26-b52e-3aee47a5eb97" />
This check has not been removed

Reply:
> why remove the `requiredSchema.nonEmpty` check.

Hi, I have rolled back these changes, I have kept the original logic as much as possible, and only changed the logic for creating representative files